core/src/test/java/org/projectnessie/cel/common/types/ProviderTest.java

```diff
@@ -87,12 +88,29 @@ void typeRegistryCopy() {
   void typeRegistryEnumValue() {
     ProtoTypeRegistry reg = newEmptyRegistry();
     reg.registerDescriptor(GlobalEnum.getDescriptor().getFile());
+    reg.registerDescriptor(OutOfOrderEnumOuterClass.getDescriptor().getFile());
 
     Val enumVal = reg.enumValue("google.api.expr.test.v1.proto3.GlobalEnum.GOO");
-    assertThat(enumVal).extracting(Val::intValue).isEqualTo((long) GlobalEnum.GOO.ordinal());
+    assertThat(enumVal).extracting(Val::intValue).isEqualTo((long) GlobalEnum.GOO.getNumber());
 
     Val enumVal2 = reg.findIdent("google.api.expr.test.v1.proto3.GlobalEnum.GOO");
     assertThat(enumVal2.equal(enumVal)).isSameAs(True);
+
+    // Previously, we checked `getIndex` on the `EnumValueDescriptor`, which is the same as the
+    // `ordinal` value on the enum.
+    assertThat(OutOfOrderEnumOuterClass.OutOfOrderEnum.TWO.ordinal())
+        .isEqualTo(OutOfOrderEnumOuterClass.OutOfOrderEnum.TWO.getValueDescriptor().getIndex());
+    // Test the case where the protobuf-defined value for the enum differs from the generated Java
+    // enum's ordinal() function.
+    assertThat(OutOfOrderEnumOuterClass.OutOfOrderEnum.TWO.getNumber())
+        .isNotEqualTo(OutOfOrderEnumOuterClass.OutOfOrderEnum.TWO.ordinal());
+    assertThat(OutOfOrderEnumOuterClass.OutOfOrderEnum.TWO.getNumber())
```

Comment:
Nit: you can add the 2nd `.isEqualTo()` directly behind the one in line 106 (aka: remove the semicolon in 106 and remove line 107).